src/api/index.test.tsx

```diff
@@ -0,0 +1,44 @@
+import { HttpMethod, callApi } from '.';
+import { initialState as apiState } from '../reducers/api';
+
+describe(__filename, () => {
+  describe('callApi', () => {
+    it('calls the API', async () => {
+      fetchMock.mockResponse(JSON.stringify({}));
+
+      await callApi({ apiState, endpoint: '/foo/' });
+
+      expect(fetch).toHaveBeenCalledWith(`/api/v4/foo/`, {
+        method: 'GET',
+        headers: {},
+      });
+    });
+
+    it('adds a trailing slash to the endpoint if there is none', async () => {
+      fetchMock.mockResponse(JSON.stringify({}));
+
+      await callApi({ apiState, endpoint: '/foo' });
+
+      expect(fetchMock.mock.calls[0][0]).toEqual(`/api/v4/foo/`);
```

Comment:
Aww, this won't be as nice as `sinon.assert.calledWith()`. I like the sinon matchers. I guess the typescript and jest integration is nice with your approach, though.

Reply:
Yep